dev/TreeView/ViewModel.cpp

```diff
@@ -26,27 +26,33 @@ class SelectedTreeNodeVector :
     Implement_Vector_Read(SelectedTreeNodeVectorOptions)
 
 private:
-    ViewModel* m_viewModel{ nullptr };
+    winrt::weak_ref<winrt::DependencyObject> m_viewModel{ nullptr };
     void UpdateSelection(winrt::TreeViewNode const& node, TreeNodeSelectionState state)
     {
         if (winrt::get_self<TreeViewNode>(node)->SelectionState() != state)
         {
-            m_viewModel->UpdateSelection(node, state);
-            m_viewModel->NotifyContainerOfSelectionChange(node, state);
+            if (auto viewModel = m_viewModel.get().try_as<ViewModel>())
```

Comment:
If the get() fails (which it can because it's a weak ref) won't this null deref?
I see you stored the weak_ref as a DO instead of ViewModel. Did it not work to store as a ViewModel weak_ref?

Reply:
If you can't make a weak_ref to ViewModel then consider making a private GetViewModel() method that returns a properly casted ViewModel or null if the ref expired. Then you don't have to spread the checks through every place you have here that uses m_viewModel.

Reply:
weak_ref<ViewModel> didn't work for some reason :\
Added GetViewModel()

Reply:
What does "didn't work" mean?

Reply:
make_weak(viewModel) throws a build error
```
1>d:\microsoft-ui-xaml\buildoutput\intermediates\microsoft.ui.xaml\obj\cppwinrt\platform\winrt\base.h(4184): note: No constructor could take the source type, or constructor overload resolution was ambiguous
1>d:\microsoft-ui-xaml\dev\treeview\viewmodel.cpp(57): note: see reference to function template instantiation 'winrt::weak_ref<D> winrt::make_weak<ViewModel>(const T &)' being compiled
1>        with
1>        [
1>            D=ViewModel,
1>            T=ViewModel
1>        ]
```

Need to implement `IWeakReferenceSource` on ViewModel to make it work?

Reply:
ViewModel is a C++/WinRT implemented object so it should pick up IWeakReferenceSource from there. The error says that the overload resolution was ambiguous -- were there any more output messages that say what the ambiguity was?

Reply:
Talked offline, using get_weak and removing the const keyword in function parameter fixed it.